### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by charlottekostelic**
Changed:
- changed Retry args in `MetadataSession` and `WorldcatSession` to camel case (all args should now be in camel case per #17) 
- updated links to OCLC documentation/website per #57
- split `index.md` into multiple files to allow for multiple pages on https://bookops-cat.github.io/bookops-worldcat (new pages listed below)
- theme of docs: `readthedocs` > `material`
- typos in multiple files (I added the Code Spell Checker extension in VS Code which identified a bunch of typos)
- `target-version` for black in `pyproject.toml`

Added:
- `bytes` and `BytesIO` as types for `MetadataSession` methods that send MARC record in request body
- `mkdocs-material` as dev-dependency
- `extra.css` to customize theme with NYPL colors
- Separate pages in documentation for different functionality:
  - Get Started (`start.md`)
  - Search WorldCat (`search.md`)
  - Manage Bibliographic Records (`manage_bibs.md`)
  - Manage Institution Holdings (`manage_holdings.md`)
  - Search and Manage Local Data (`local.md`)
  - Advanced Usage (`advanced.md`)
- Draft contribution guidelines per #86 (`contributing.md`, there's still more to add here in a future PR)
- how to add RegistryID as context in `WorldcatAccessToken` per #83 (this is in the Advanced Usage > WorldcatAccessToken section)
- 1.0 to changelog
- sections in README and docs on changes in version 1.0
- `py.typed` file and updated `pyproject.toml`

Removed:
- content from `tutorials.md`. The page is there as a placeholder and I will update it in a future PR.

### CODE REVIEW

File: docs/contributing.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# How to Contribute
```
Comment (klinga) [inline docs/contributing.md:1]: Looks great!

File: docs/index.md [hunk 2]
```diff
@@ -2,517 +2,248 @@
 
 [![Build Status](https://github.com/BookOps-CAT/bookops-marc/actions/workflows/unit-tests.yaml/badge.svg?branch=master)](https://github.com/BookOps-CAT/bookops-worldcat/actions) [![Coverage Status](https://coveralls.io/repos/github/BookOps-CAT/bookops-worldcat/badge.svg?branch=master&service=github)](https://coveralls.io/github/BookOps-CAT/bookops-worldcat?branch=master) [![PyPI version](https://badge.fury.io/py/bookops-worldcat.svg)](https://badge.fury.io/py/bookops-worldcat) ![PyPI - Python Version](https://img.shields.io/pypi/pyversions/bookops-worldcat) [![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black) [![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg)](https://opensource.org/licenses/MIT)
```
Comment (klinga) [inline docs/index.md:3]: master > main in links for various badges

File: docs/index.md [hunk 3]
```diff
@@ -2,517 +2,248 @@
...
+|get brief bib resource| `get_brief_bib` | `brief_bibs_get` |
+|get full bib record| `get_full_bib` | `bib_get` |
+|get current holdings for record| `holding_get_status` | `holdings_get_current` |
+|set holdings on a record| `holding_set` | `holdings_set` |
+|unset holdings on a record| `holding_unset` | `holdings_unset` |
+|search for brief bibs| `search_brief_bibs` | `brief_bibs_search` |
+|get other editions of brief bibs| `search_brief_bib_other_editions` |`brief_bibs_get_other_editions` |
+|get current oclc number| `search_current_control_numbers` | `bib_get_current_oclc_number` |
+|search member holdings| `search_general_holdings` | `summary_holdings_search` |
+|search shared print holdings| `search_shared_print_holdings` | `shared_print_holdings_search` |
```
Comment (klinga) [inline docs/index.md:236]: This is really nice!

File: docs/search.md [hunk 4]
```diff
@@ -0,0 +1,430 @@
...
+    "mostPopular": [
+      "PQ4866.E6345 A8113 2012"
+    ]
+  }
+}
+```
+## Advanced Search Functionality
+
+!!! info
+    While most arguments passed to `/search/` endpoints (such as `brief_bibs_search`, `local_bibs_search`, and `summary_holdings_search`) are joined using the 'AND' operator, when both `itemType` and `itemSubType` are used in a query, they are joined using the 'OR' operator.
```
Comment (klinga) [inline docs/search.md:363]: 👍

**Review by klinga — APPROVED** [review]
@charlottekostelic You took the documentation to entirely another level 👏 
I'm looking forward to seeing these published and in action. I see a lot of mkdocs-material markdown syntax that 'm super curious to experience.
Have only a minor suggestions/questions.

File: bookops_worldcat/metadata_api.py [hunk 5]
```diff
@@ -152,7 +152,7 @@ def _url_search_lbd(self) -> str:
 
     def bib_create(
         self,
-        record: str,
+        record: Union[str, bytes, BinaryIO],
```
[the remark this answers is not in the document]
Comment (charlottekostelic) [reply bookops_worldcat/metadata_api.py:155]: `bib_match` and `bib_validate` work when passed a `str` but I still haven't tested `bib_create` or `bib_replace`.

**Pull request merged**